pydantic/type_adapter.py

```diff
@@ -204,7 +204,7 @@ def __init__(
         except AttributeError:
             if module is None:
                 f = sys._getframe(1)
-                module = cast(str, f.f_globals['__name__'])
+                module = cast(str, f.f_globals.get('__name__', ''))
```

Comment:
Do we want to default to `'__main__'` instead of `''`?

Reply:
hmm, not sure. @samuelcolvin do you have any opinion here?